hathor/manager.py

```diff
@@ -182,6 +183,22 @@ def start(self) -> None:
         """
         self.log.info('Starting HathorManager...')
         self.log.info('Network: {network}', network=self.network)
+        # If it's a full verification, we save on the storage that we are starting it
+        # this is required because if we stop the initilization in the middle, the metadata
+        # saved on the storage is not reliable anymore, only if we finish it
+        if self._full_verification:
+            self.tx_storage.enable_full_verification()
+        else:
+            # If it's a fast initialization and the last time a full initialization stopped in the middle
+            # we can't allow the full node to continue, so we need to remove the storage and do a full sync
+            # or execute an initialization with full verification
+            if self.tx_storage.running_full_verification_active():
+                self.log.error(
```

Comment:
We should also mark as `full_verification_required` if the full node is not properly shutdown.

Reply:
How is that possible? If it's not properly shut down, we can't detect and take any action before it's shut down, right?

Reply:
We can have a flag `not-properly-shutdown` set to true. During the shutdown, after the finish our tasks, we mark it as false.

Reply:
Another very common way to do it is to create an empty file `running` that is deleted during the shutdown.

Reply:
@msbrogli would something like this work for you? https://github.com/HathorNetwork/hathor-core/pull/89/commits/bdb68cd711b783cfbafb08f4d20848c2e66b9dcf

Reply:
It seems good! In the explanation, I'd add that the metadata is the only part that may be wrong (not the blocks / transactions themselves).

Reply:
Done